### ISSUE TITLE
Date Taken Overlay

### ISSUE THREAD

**Issue by bobbytables2048** [theothernt/AerialViews#144]
For media over a network share, it would be good to have an overlay option to show the file date or date taken for photos and videos.

**Comment by theothernt** [comment]
This is something I hope to add in future - what's likely to come first is support for the EXIF data in photos.

But regarding the date take or file date, would you expect that to be a separate overlay or just an option in the Location/Description overlay?

**Comment by theothernt** [comment]
The current thinking is...

I have a new Overlay called "Photo metadata", it will allow you to pick any two pieces of data/text from the EXIF data inside the photo.

The existing Overlay, Location/Description, will be limited to the something like XPTitle or ImageDescription - as I think that's what most people will use.

**Comment by bobbytables2048** [comment]
Hi - I think that would work for my use case. Happy to provide feedback for a beta version when it is ready.

**Comment by 00-dave** [comment]
Just reading this - I set up and was able to get going as a Default Screen Saver (wish Google Chrome Cast made that easier) - I wasn't able to expose EXIF data, I think that would be a great add on - Even better, would be to use the GPS data from the EXIF and resolve a location (Country/City) - Footo does something similar - theoretically I can use another tool to manually write the EXIF data in to each file, so the EXIF is a critical first step. (I'm sourcing photos from a SMB NAS).

**Comment by nation-wide** [comment]
Any ETA on this? I would really like to have "Date taken" visilbe screen. Thanks!

**Comment by theothernt** [comment]
> Any ETA on this? I would really like to have "Date taken" visilbe screen. Thanks!

I'm hoping for this feature to be in the next couple of releases, probably v1.8.0 - but I expect it to be in the Play Store public beta in about 2 weeks - assuming next week's release goes ok.

**Comment by christerdk** [comment]
This would be great for media from an Immich source, too 👍🏻

**Comment by theothernt** [comment]
Taking a lot longer than expected as I'm rewriting how the Location / Description / Metadata overlays works.

You'll be able to assign different pieces of metadata depending on what's available in the photo or video. For example...

a) An Apple video with points of interest (POIs)
b) A photo from Immich with a description, but no location data
c) A photo from Immich with location data, but no description

For that overlay, I would set the following configuration:

1. Apple videos - Points of Interest
2. Immich media - Description
3. Immich media - Location

So one overlay can display metadata for various types of media. Also, the order will be important. So in the above example, the Description is shown if there is both Description and Location data.

I hope that makes sense!

**Comment by nation-wide** [comment]
But what about the original question about "Date taken" property? I would like to just show that date as opposed to today's date.

For example like below. Thanks.

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/5e179c71-b9bd-4879-9535-9048457b09c2" />

**Comment by theothernt** [comment]
> But what about the original question about "Date taken" property? I would like to just show that date as opposed to today's date.

As mentioned above, it's a lot of work and not just the date metadata from a photo. It won't be in v1.8.0 but I hope to have something in v1.8.0 beta1.

**Comment by GWAVE23** [comment]
Waiting for this update. Will be a great addition and exactly what I'm looking for!

**Comment by pzjuls** [comment]
Yes I'm loving this app well done [theothernt] but really need Date taken and location to work how do I get the Beta version to test?
Thanks Pzjuls

**Comment by theothernt** [comment]
Ok, v1.8.1 is out so this feature will be in v1.8.2 - it should be in the public betas first.

I have done some initial testing with a Samba share, loading images, showing EXIF data and it performs well.

The UI for this will give people a ton of new options, so I hope it won't be too confusing.

**Comment by pzjuls** [comment]
This is amazing news, Neil—well done! Will this update also include GPS location data?

I’m currently running the beta version on a Gen 1 Shield via network share. I noticed my system hasn't updated since around January 9th. Is there a way to check what the current "official" beta version is versus what I’m running?

Also, do I need to force an update, or has there simply not been a new upload since the 9th? Thanks!

**Comment by theothernt** [comment]
> This is amazing news, Neil—well done! Will this update also include GPS location data?

Photos contain EXIF data. Part of the EXIF data is the GPS coordinates - not a friendly location like "Dublin, Ireland". Android has a quick look up feature which turns 37.7749, -122.4194 into San Francisco, for example.

> I’m currently running the beta version on a Gen 1 Shield via network share. I noticed my system hasn't updated since around January 9th. Is there a way to check what the current "official" beta version is versus what I’m running?

The current beta is v1.8.1 beta13 - you'll have to wait for 1.8.2 beta1 or beta2. It might be a week or two.

**Comment by pzjuls** [comment]
Thanks for the update, Neil! Regarding future features: would "friendly" GPS location data ever be possible down the line?

**Comment by theothernt** [comment]
That will be part of the feature.

I initially worried about the performance of a) opening the image for metadata b) opening the image again to display it and c) lookup the location based on GPS coordinates - but it appears to work well.

**Comment by mblaser79** [comment]
> Ok, v1.8.1 is out so this feature will be in v1.8.2 - it should be in the public betas first.

First off, this thread was awesome news to hear, it's the last thing I needed to make this the perfect screensaver. So thank you for that.

Second, I just got the 1.8.2 beta on my Onn device, should I be seeing these new overlays or are they not live yet? I had a look around the settings and didn't see anything about them.

**Comment by theothernt** [comment]
> Second, I just got the 1.8.2 beta on my Onn device, should I be seeing these new overlays or are they not live yet? I had a look around the settings and didn't see anything about them.

It's not out yet, I'll update the thread when it is.

**Comment by theothernt** [comment]
Ok, it's nearly here (finally!) - just a few more days.

I need to do some more testing and some text changes. Also, you can have more than 1 metadata overlay.

<img width="965" height="535" alt="Image" src="https://github.com/user-attachments/assets/eafca27b-e727-4f11-abc2-92a25e6e1a5a" />

**Comment by pzjuls** [comment]
> Ok, it's nearly here (finally!) - just a few more days.
> 
> I need to do some more testing and some text changes. Also, you can have more than 1 metadata overlay.
> 
> <img alt="Image" width="965" height="535" src="https://private-user-images.githubusercontent.com/136865/552886378-eafca27b-e727-4f11-abc2-92a25e6e1a5a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzE3NTU3OTQsIm5iZiI6MTc3MTc1NTQ5NCwicGF0aCI6Ii8xMzY4NjUvNTUyODg2Mzc4LWVhZmNhMjdiLWU3MjctNGYxMS1hYmMyLTkyYTI1ZTZlMWE1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIyMlQxMDE4MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTQ0MGFhOWM4OWZhYmE2YjFlYWQ0MDk3NGQ2MWQ3ZWUzY2RlNmIwOTQyZmMzNGQxZjNiYjllY2QxZmYxZTY5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.5qDzFp8QCXCVKbjB2Jnu4v_21070v5FLTsWbcRQM38E">

Amazing news well done Neil 👌

**Comment by theothernt** [comment]
That feature is now live in the public beta in the Play Store.

There is a performance issue but I have submitted a fix, so that should be live in s few hours.

**Comment by mblaser79** [comment]
I tested the new update out today and it works perfectly. Amazing work!

**Comment by theothernt** [comment]
> I tested the new update out today and it works perfectly. Amazing work!

That's great to hear. 

If you use the new metadata overlay with all of its options, let me know if its works for you or not. Do all of the options make sense, etc?

Also, does anyone want a beta APK or is the Play Store ok?

**Comment by AnttiKaleva** [comment]
> > I tested the new update out today and it works perfectly. Amazing work!
> 
> That's great to hear.
> 
> If you use the new metadata overlay with all of its options, let me know if its works for you or not. Do all of the options make sense, etc?
> 
> Also, does anyone want a beta APK or is the Play Store ok?

Please share the beta APK as I don't see it in Play Store. Thanks very much for this feature! 🥇

**Comment by theothernt** [comment]
> Please share the beta APK as I don't see it in Play Store. Thanks very much for this feature! 🥇

You can sign up here on a PC, Mac or phone... https://play.google.com/apps/testing/com.neilturner.aerialviews

Once opted-in, the beta usually appears in the Play Store within an hour or so.

**Comment by pzjuls** [comment]
I tested it very breathily on Sunday and it is working great thanks Neil amazing 👏👏👏👏
I do need to do some more testing but not all the pictures were coming up with the location and some only had like the county the picture was takin in I will do more testing and report back.
But its amazing we love it 👌

**Comment by AnttiKaleva** [comment]
> > Please share the beta APK as I don't see it in Play Store. Thanks very much for this feature! 🥇
> 
> You can sign up here on a PC, Mac or phone... https://play.google.com/apps/testing/com.neilturner.aerialviews
> 
> Once opted-in, the beta usually appears in the Play Store within an hour or so.

Perfect. Tested and works really well.

**Comment by mysweethome-afk** [comment]
Neil,
first of all, thank you very much for this great app! Now I can finally say "bye bye Google Photos"! The beta works like a charm and the exif data of the Pictures on my samba share are well shown!

I have one Question and a little Feedback to aspect ratio.

Question:
Do you plan to integrate more than 2 metadata overlays in the app? It would be great if there were more than two. I want to show Folder Name, Date taken and  Description at any picture. But within the beta I only can show two of them. 

Feedback aspect ratio:
In the beta there is a little problem by using Portraits with ratio 3:4. I have compared the official and the beta release. 
In the official Version, Pictures with this ratio are shown well with the Setting "Scale without crop" (see [Picture 1](https://github.com/user-attachments/assets/e30bef48-7843-47b4-b7c3-8eda352365f3)). 
In the beta the 3:4 Pictures are not cropped properly (see [Picture 2](https://github.com/user-attachments/assets/fe83fc9d-8b0a-4721-b977-ac46fdef7639)). 
If you have Portraits with 9:16, everthing is fine in both Version (see [Picture 3](https://github.com/user-attachments/assets/4ae99a19-1c20-47dc-9eb6-10e48fd66e9a)).


Best Regards
Stefan

**Comment by theothernt** [comment]
> Question: Do you plan to integrate more than 2 metadata overlays in the app? It would be great if there were more than two. I want to show Folder Name, Date taken and Description at any picture. But within the beta I only can show two of them.

I will add 2 more, so there is a total of 4.
 
> Feedback aspect ratio: In the beta there is a little problem by using Portraits with ratio 3:4.

Thanks for reporting that issue. I should have a fix for that in the next 12 hours.

**Comment by mysweethome-afk** [comment]
Neil, again great job! Thank you! The aspect ratio is fine now! :-) And also thank you for the further metadata fields! 

I have a little german problem with EXIF-Description and umlauts/additional character. These are shown with ???. The EXIF-Location shows the right umlauts, for example "Rüdesheim" and if I use "folder name" it is also ok. Is the problem in EXIF-Description a problem from the EXIF-Data, from the charset of the Description field or comes it from android?

At the moment I have to use english language in the app to see all the configuration from the metadata. Is this the problem?

I have try to mask the umlauts with `&ouml;` and the other options, but it only show "???"

And another little question: what is the meaning of "Dynamic Points of Interests" in metadata Videos?

**Comment by theothernt** [comment]
> I have a little german problem with EXIF-Description and umlauts/additional character. These are shown with ???. The EXIF-Location shows the right umlauts, for example "Rüdesheim" and if I use "folder name" it is also ok. Is the problem in EXIF-Description a problem from the EXIF-Data, from the charset of the Description field or comes it from android?

The problem with the characters in the EXIF data is down a character encoding issue. I'm sure I can fix that.

> At the moment I have to use english language in the app to see all the configuration from the metadata. Is this the problem?

Can you give me more detail about this issue, I'm not sure I understand - are you seeing different menu options between German and English languages?

> And another little question: what is the meaning of "Dynamic Points of Interests" in metadata Videos?

As for "Dynamic Points of Interest - they are a list of longer descriptions that change while the video is playing. Only the Apple videos have these at the moment.

**Comment by mysweethome-afk** [comment]
> The problem with the characters in the EXIF data is down a character encoding issue. I'm sure I can fix that.

Thank you!

>Can you give me more detail about this issue, I'm not sure I understand - are you seeing different menu options between German and English languages?

Yes, I can and yes  I do. My system locale is german. If I choose system locale or german as language setting for Menu text locale,  I only see this information, when I want to edit metadata fields:
<img width="807" height="504" alt="Image" src="https://github.com/user-attachments/assets/aa712c10-9436-4b9d-a62a-66146b97dfe1" />

If I use english for Menu text locale, I see all the fields:
<img width="789" height="571" alt="Image" src="https://github.com/user-attachments/assets/301f7569-8d1f-4e38-b3a5-1cd829ccf919" />

If the fields are configured, they will be shown, no matter which language is chosen. But if you want to edit the settings, you can only  do this with switching to english language.

**Comment by theothernt** [comment]
Thanks for that, it looks like some of the translations were out of sync for 5 languages including German.

A fix for that the and the EXIF description language issue has been submitted to the Play Store.

**Comment by mysweethome-afk** [comment]
Neil, thanks again! Looks perfect!!!
Addiotinal Characters are well shown:
<img width="843" height="465" alt="Image" src="https://github.com/user-attachments/assets/ce1d7732-0cce-47a7-963b-309e99ff31fb" />
<img width="811" height="452" alt="Image" src="https://github.com/user-attachments/assets/5557d8d5-03a8-42f0-9c1e-4c5a433e2ebc" />

And the  translations are also done, the fields are now available in other languages, not only in english.

Thank you very much!

**Comment by theothernt** [comment]
Fantastic! Thanks for finding the issues and doing all that testing!

**Comment by theothernt** [comment]
No issues are being reported by anyone using the beta so I'll probably make a release in the next few days!

**Comment by mysweethome-afk** [comment]
Probaly not a bug, but two questions to the Video metadata fields in case of use for local videos.

1. 
There are these options to use:
<img width="494" height="263" alt="Image" src="https://github.com/user-attachments/assets/fbe011aa-dbba-4766-a87f-94a48ad723b8" />
I have tested with a video and have put in several fields some information:
<img width="326" height="231" alt="Image" src="https://github.com/user-attachments/assets/fce65569-7605-4a1b-82ae-789d8cb23fde" />

None of them are shown in the video. Especially the data in the field Short Description I had been expected. What kind of function has this field in the app? How can I use this?

2. 
Is there any chance to show/use Date taken in the videos? In the metadata there is the information for Date taken available, but this is not an EXIF data i know.

Best Regards
Stefan

**Comment by theothernt** [comment]
> None of them are shown in the video. Especially the data in the field Short Description I had been expected. What kind of function has this field in the app? How can I use this?

For the moment, the Short Description and POIs are for Apple, Amazon, etc videos only.

I may have to add more notices to make it clearer for people.

> Is there any chance to show/use Date taken in the videos? In the metadata there is the information for Date taken available, but this is not an EXIF data i know.

Not at the moment as trying to get EXIF data from videos is messy, there is no standard way. Android does have an API for this so I will test this in future but MKV, MP4, MOV, etc all handle things differently.

**Comment by mysweethome-afk** [comment]
> >  For the moment, the Short Description and POIs are for Apple, Amazon, etc videos only.
> 
This means it works only when the videos are hosted by these providers? Or how are the dependencies for that?

> >Not at the moment as trying to get EXIF data from videos is messy, there is no standard way. Android does have an API for this so I will test this in future but MKV, MP4, MOV, etc all handle things differently.

For the beginning it could be a good thing to integrate some single codecs/container with defined specs. All of them at once are surely damn much work... :-/

**Comment by theothernt** [comment]
> This means it works only when the videos are hosted by these providers? Or how are the dependencies for that?

Yes, there is a 'manifest' file which lists all the videos, various qualities, a short title and POI info. In some cases, this was added manually. For example...

https://raw.githubusercontent.com/theothernt/AerialViews/refs/heads/master/app/src/main/res/raw/tvos15.json

> For the beginning it could be a good thing to integrate some single codecs/container with defined specs. All of them at once are surely damn much work... :-/

I agree. But what I will do is research how, if possible, to get EXIF type data from the most popular formats.

**Comment by mysweethome-afk** [comment]
Ok, sounds interesting with the manifest file. But I'm afraid that its not so easy that I only have to put the manifest file in the videos folder with the needed information for short title an further informations.  Isn't?

**Comment by theothernt** [comment]
> Ok, sounds interesting with the manifest file. But I'm afraid that its not so easy that I only have to put the manifest file in the videos folder with the needed information for short title an further informations. Isn't?

It's only for the built-in videos. If I am unable to automatically get meta/exif data from videos in future, I will add a feature to read extra video info from a text file.

**Comment by mysweethome-afk** [comment]
Neil, that sounds good! And this could be a good way to avoid problems with the variety of video codecs. I'm curious about that! :-)

**Comment by theothernt** [comment]
> Is there any chance to show/use Date taken in the videos? In the metadata there is the information for Date taken available, but this is not an EXIF data i know.

A quick update - it appears the video player I use, ExoPlayer, actually extracts all the metadata we want to show, so it should be easy enough to add.

I need to do some testing but if it works, the feature will be ready to test in the next public beta due later in the week.

Ref: Player Listener onMediaMetadataChanged vs MediaMetadataRetriever

**Comment by theothernt** [comment]
This is rolling out now in the Play Store (v1.8.2)

**Comment by theothernt** [comment]
If anyone wants to help with testing and feedback of this feature for video files from local storage, Samba, etc - please head over to #337
